apiserver/stateauthenticator/modeluser.go

```diff
@@ -177,7 +178,7 @@ func (u *modelUserEntity) UpdateLastLogin() error {
 	if u.user != nil {
 		err1 := u.user.UpdateLastLogin()
 		if err == nil {
```

Comment:
This is a difficult one to read, as the logic is a bit backwards. I might re-look at this one, so the readability improves as first glance it's not quite a clear cut as it could be.